src/mavsdk/core/mavsdk_impl.cpp

```diff
@@ -119,6 +119,49 @@ std::vector<std::shared_ptr<System>> MavsdkImpl::systems() const
     return systems_result;
 }
 
+std::optional<std::shared_ptr<System>> MavsdkImpl::first_autopilot(double timeout_s)
+{
+    {
+        std::lock_guard<std::recursive_mutex> lock(_systems_mutex);
+        if (!_systems.empty()) {
+            return {_systems[0].second};
+        }
+    }
+
+    if (timeout_s == 0.0) {
```

Comment:
Shouldn't this be `<= 0.0`? Would make sense to me :thinking:

Reply:
A negative timeout should wait forever.

Reply:
Oh, I had missed that :see_no_evil: